crates/red_knot_python_semantic/src/types.rs

```diff
@@ -261,12 +262,12 @@ impl<'db> UnionType<'db> {
     }
 }
 
-struct UnionTypeBuilder<'db> {
+struct UnionBuilder<'db> {
```

Comment:
It might be helpful to turn some of your PR comments into a document. For example, that a union builder isn't guaranteed to return a union. It might be useful to extract the builders into their own `types/builder` module so that you can make your PR comment a module level comment.